lib/manager/gitlabci/extract.ts

```diff
@@ -79,3 +82,49 @@ export function extractPackageFile(content: string): PackageFile | null {
   }
   return { deps };
 }
+
+export async function extractAllPackageFiles(
+  config: ExtractConfig,
+  packageFiles: string[]
+): Promise<PackageFile[] | null> {
+  const filesToExamine = new Set<string>(packageFiles);
+  const results: PackageFile[] = [];
+
+  // extract all includes from the files
+  while (filesToExamine.size > 0) {
+    const file = filesToExamine.values().next().value;
+    filesToExamine.delete(file);
+
+    const content = await readLocalFile(file, 'utf8');
+    const doc = yaml.safeLoad(content, { json: true }) as any;
+    if (doc?.include && is.array(doc.include)) {
+      for (const includeObj of doc.include) {
+        if (includeObj.local) {
+          const fileObj = (includeObj.local as string).replace(/^\//, '');
+          if (!filesToExamine.has(fileObj)) {
+            filesToExamine.add(fileObj);
+          }
+        }
+      }
+    }
+
+    const result = extractPackageFile(content);
```

Comment:
@HonkingGoose Please open a new issue:

Refactor `extractPackageFile` to use `js-yaml` and reuse already parsed object from `extractAllPackageFiles`.
So we should extract a new function `parseDoc` to be reused by both extract functions.

Reply:
Done, see #9115. 😉